Woodoku/src/ui/command_line_ui.py

```diff
@@ -1,29 +1,88 @@
 from typing import List, Tuple
 
+from elements import *
 from entity.woodoku_board import WoodokuBoard
 from entity.woodoku_shape import WoodokuShape
 from ui.ui_interface import UIInterface
 
+NUM_SHAPES = 3
+
 
 class CommandLineUI(UIInterface):
 
     def show_start_game(self, board: WoodokuBoard) -> None:
-        pass
+        print(orange("\nWelcome to Wooduku"))
+        print(board)
 
     def show_board(self, board: WoodokuBoard) -> None:
-        pass
+        print(board)
 
     def choose_shape(self, shapes: List[WoodokuShape], shape_availability: List[bool]) -> int:
-        pass
+        shapes_str = ""
+        for idx, avai in enumerate(shape_availability):
+            if avai:
+                shapes_str += f"{idx}.\n{str(shapes[idx]).rstrip()}\n\n"
+        print(shapes_str)
+        val = ""
+        while not isinstance(val, int) or val not in range(NUM_SHAPES):
+            val = input(f"Please choose one of the {orange('shapes')}:\n")
+            try:
+                val = int(val)
+                if val not in range(NUM_SHAPES):
+                    raise ValueError
+            except ValueError:
+                print(red("Please choose an integer value from 0 to 2. Try again"))
+        return val
 
     def put_shape_at(self) -> Tuple[int, int]:
-        pass
+        confirmed = False
+        while not confirmed:
+            x = ""
+            y = ""
+            while not isinstance(x, int):
+                try:
+                    x = int(input(f"Enter a {orange('x coordinate')} on a 9X9 board:\n"))
+                except ValueError:
+                    print(red("Please enter an integer x coordinate. Try again\n"))
+            while not isinstance(y, int):
+                try:
+                    y = int(input(f"Enter a {orange('y coordinate')} on a 9X9 board:\n"))
+                except ValueError:
+                    print(red("Please enter an integer y coordinate. Try again"))
```

Comment:
Why not getting both x and y from user at once?

Reply:
We could get them at once, but after trying both implementations. I found that entering one coordinate at a time is fast er since they don't need to enter any separator such as space or comma. We could change if you think getting both x and y at once is better

Reply:
Fair point.